handbook/engineering/cloud/goals.md

```diff
@@ -11,7 +11,7 @@
 1. ✅ The Sourcegraph team members can add public code from GitHub.com and GitLab.com.
 1. 🔄 The Sourcegraph team members can add private code from Github.com and GitLab.com to Sourcegraph Cloud.
 1. 🔄 Any user can add public code from GitHub.com and GitLab.com.
-1. Sourcegraph Cloud is Generally Available (GA) for users to add private code from GitHub.com and GitLab.com.
+1. Sourcegraph Cloud is Generally Available (GA) for users to add private code from GitHub.com and GitLab.com. ETA 2021-02-01.
```

Comment:
Thanks for this detailed breakdown!

It looks like our bottleneck in milestone 3 comes down to frontend implementation. That makes sense, given our newer teammates are still ramping up.

I believe contexts can be shifted into milestone 4, as it's not essential for the milestone to **work**. It will be essential to reach a baseline usability experience for GA. However, that doesn't address the bigger picture of completing Milestone 4 by end of quarter, unless the contexts project can be wholly owned by the Search team and delivered within that timeline.

Notifications UX and background processes are part of the same project, and I'm exploring it right now. It's something we can easily keep to limited functional scope (make system status visible) so that it's feasible in the short term.

Is there anything else in here that either isn't wholly required to make it work for milestone 4?

Reply:
> Add search contexts

@christinaforney and I have tentatively inserted this in the Search roadmap starting at some point in January. With that in mind, and given other ongoing work in the Search team's roadmap, this would _optimistically_ ship by 2021-02-01.

Reply:
In that case, I think it's fair to say that contexts are part of Milestone 4. @tsenart I've moved it in Productboard to reflect this.

Reply:
> Cloudflare web application firewall — I believe this is done, can you confirm @ElizabethStirling?

Yep, Cloudflare WAF is enabled